### PR TITLE
fix(ivy): reset style property value defined using [style.prop.px]

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Prior to this change, setting style prop value to undefined or empty string would not result in resetting prop value in case the style prop is defined using [style.prop.px] syntax. The problem is that the check for empty value (and thus reseting the value) considered successful only in case of `null` value. This commit updates the check to use `isStylingValueDefined` function that also checks for undefined and empty string.

This PR resolves FW-1676.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/test/acceptance/styling_spec.ts [hunk 1]
```diff
@@ -2310,6 +2310,37 @@ describe('styling', () => {
...
+    TestBed.configureTestingModule({declarations: [MyComp]});
+    const fixture = TestBed.createComponent(MyComp);
+    fixture.detectChanges();
+
+    const div = fixture.nativeElement.querySelector('div');
+
+    const expected = ['', '', '', '20px'];
+    const buttons = fixture.nativeElement.querySelectorAll('button') !;
+    expect(div.style.left).toBe('10px');  // initial value
+    Array.from(buttons).forEach((button: any, index: number) => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/styling_spec.ts]: do we need all those `<button>` elements in this test? Given that we call `fixture.detectChanges()` on each and every `click`, I would rather:
* remove all the `<button>` elements from a template
* set the `left` on the fixture.ComponentInstance directly.

**Review by pkozlowski-opensource — CHANGES_REQUESTED** [review]
Thnx for the fix @AndrewKushnir - the fix itself LGTM but I think that we can simplify a test (see in-line comment)

**Comment by pkozlowski-opensource** [comment]
@AndrewKushnir wouldn't your change also solve https://github.com/angular/angular/issues/33650 ? 
Maybe we should add a test for the `[ngStyle]` case as well?

File: packages/core/test/acceptance/styling_spec.ts (hunk 1 above)
Comment (matsko) [reply packages/core/test/acceptance/styling_spec.ts]: Yes I agree. Just get the component instance and change the class members.

**Review by matsko — CHANGES_REQUESTED** [review]

**Comment by AndrewKushnir** [comment]
Thanks for review @matsko and @pkozlowski-opensource!

I was modeling the test after the use-case where the problem was originally found, but I do agree that it's a bit too complicated and we can make it simpler :) I've updated the test as you proposed, could you please have a look?

Thank you.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](http://test/OCL:280233293:BASE:280233347:1573671368527:83d9075)
- [Ivy Presubmit](http://test/OCL:280233293:BASE:280233496:1573671403352:bed7d951)

**Comment by AndrewKushnir** [comment]
@pkozlowski-opensource sorry I missed your question related to #33650. I just tested the use-case with `[ngStyle]` and the changes in this PR and the problem still exists (so it requires additional investigation and fixed in a separate PR). Thank you.

**Review by matsko — APPROVED** [review]

**Pull request closed without merging**